lib/Dialect/Triton/IR/Ops.cpp

```diff
@@ -438,6 +438,31 @@ OpFoldResult MakeRangeOp::fold(FoldAdaptor adaptor) {
   return {};
 }
 
+LogicalResult MakeRangeOp::verify() {
```

Comment:
This is quite a nit but an alternative is to add `DeclareOpInterfaceMethods<InferTypeOpInterface>` and infer the type from the attributes. 
That being said I'm not sure whether this is better and probably not worth your time.

Reply:
Ah that would be cool, especially if you could omit the return type in the IR.

But how does this work with encodings?  We cannot infer the correct layout for a MakeRange op.  So therefore presumably we cannot infer the type at all?

Reply:
ah you're right I missed that part.